controllers/nstemplateset/nstemplateset_controller.go

```diff
@@ -58,14 +58,14 @@ func (r *Reconciler) SetupWithManager(mgr manager.Manager, allNamespaceCluster r
 	mapToOwnerByLabel := handler.EnqueueRequestsFromMapFunc(commoncontroller.MapToOwnerByLabel("", toolchainv1alpha1.SpaceLabelKey))
 	build := ctrl.NewControllerManagedBy(mgr).
 		For(&toolchainv1alpha1.NSTemplateSet{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
-		Watches(&source.Kind{Type: &corev1.Namespace{}}, mapToOwnerByLabel).
-		Watches(source.NewKindWithCache(&rbac.Role{}, allNamespaceCluster.GetCache()), mapToOwnerByLabel, builder.WithPredicates(commonpredicates.LabelsAndGenerationPredicate{})).
-		Watches(source.NewKindWithCache(&rbac.RoleBinding{}, allNamespaceCluster.GetCache()), mapToOwnerByLabel, builder.WithPredicates(commonpredicates.LabelsAndGenerationPredicate{}))
+		Watches(&corev1.Namespace{}, mapToOwnerByLabel).
+		WatchesRawSource(source.Kind(allNamespaceCluster.GetCache(), &rbac.Role{}), mapToOwnerByLabel, builder.WithPredicates(commonpredicates.LabelsAndGenerationPredicate{})).
```

Comment:
Just curious what is new here and why replacing `Watches` with `WatchesRawSource`?

Reply:
please note that not all `Watches` are replaced with `WatchesRawSource`, only the one where there was cache. Because with the removal of Source as argument in the Watches signature, the closest replacement was `WatchesRawSource`.
If only metadata is enough, then we can use `WatchesMetadata` instead but i didn't see that to be the case, please let me know if you think otherwise. 
For reference please see this PR in controller-runtime - https://github.com/kubernetes-sigs/controller-runtime/pull/2135/files